src/wled2.ts

```diff
@@ -112,7 +114,7 @@ export = (RED: Red): void => {
           payload?.color2 ?? helpers.hexToRgb(this.config.color2),
           payload?.color3 ?? helpers.hexToRgb(this.config.color3),
         ],
-        fx: payload?.effect ?? Number(this.config.effect),
+        fx: effect != -1 ? effect : null,

```

Comment:
See my comment below on the method, but if getEffectFromString() returned null instead of -1 this could simply be `fx = effect`.